test-two-body.py

```diff
@@ -0,0 +1,60 @@
+from wrap_pyscf import *
+import unittest
+import numpy as np
+from two_body_scf_guess import *
+
+bs = "aug-cc-pvdz"
+mol = [("O", -0.167787, 1.645761, 0.108747),
+        ("H", 0.613411, 1.102620, 0.113724),
+        ("H", -0.093821, 2.209720, -0.643619),
+        ("O", 1.517569, -0.667424, -0.080674),
+        ("H", 1.989645, -1.098799, 0.612047),
+        ("H", 0.668397, -1.091798, -0.139744),
+        ("O", -1.350388, -0.964879, -0.092208),
+        ("H", -1.908991, -1.211298, 0.626207),
+        ("H", -1.263787, -0.018107, -0.055536)]
+
+H2OA = density(mol[0:3], bs)
+H2OB = density(mol[3:6], bs)
+H2OC = density(mol[6:9], bs)
+H4O2AB = density(mol[0:6], bs)
+H4O2AC = density((mol[0:3]+mol[6:9]), bs)
+H4O2BC = density(mol[3:9], bs)
+
+m,m = H2OA.shape
+n = m*2
+A = np.zeros((m*3, m*3))
+
+A[0:m, 0:m] = H4O2AB[0:m, 0:m]+H4O2AC[0:m, 0:m]-H2OA
+A[m:n, m:n] = H4O2AB[m:n, m:n]+H4O2BC[0:m, 0:m]-H2OB
+A[n:m*3, n:m*3] = H4O2AC[m:n, m:n]+H4O2BC[m:n, m:n]-H2OC
+A[0:m, m:n] = H4O2AB[0:m, m:n]
+A[m:n, 0:m] = H4O2AB[m:n, 0:m]
+A[0:m, n:m*3] = H4O2AC[0:m, m:n]
+A[n:m*3, 0:m] = H4O2AC[m:n, 0:m]
+A[m:n, n:m*3] = H4O2BC[0:m, m:n]
+A[n:m*3, m:n] = H4O2BC[m:n, 0:m]
+
+rho0 = A
+
+class TestTwoBodyGuess(unittest.TestCase):
+    def test_wtr_trimer(self):
+        mol = [("O", -0.167787, 1.645761, 0.108747),
+               ("H", 0.613411, 1.102620, 0.113724),
+               ("H", -0.093821, 2.209720, -0.643619),
+               ("O", 1.517569, -0.667424, -0.080674),
+               ("H", 1.989645, -1.098799, 0.612047),
+               ("H", 0.668397, -1.091798, -0.139744),
+               ("O", -1.350388, -0.964879, -0.092208),
+               ("H", -1.908991, -1.211298, 0.626207),
+               ("H", -1.263787, -0.018107, -0.055536)]
+        bs = "aug-cc-pvdz"
+        correct_2b = rho0
+        monomers = [mol[0:3], mol[3:6], mol[6:9]]
+        ms = monomers
+        dimers = [[(ms[0]+ms[1]), (ms[0]+ms[2])], [(ms[0]+ms[1]), (ms[1]+ms[2])], [(ms[0]+ms[2]), (ms[1]+ms[2])]]
+        tbg = main(mol, monomers, dimers, bs)
+        self.assertTrue(np.allclose(correct_2b, tbg))
+        
+if __name__ == '__main__':
+    unittest.main()
```

Comment:
Thanks for clarifying, Ryan, and thanks for the help Jonathan. Let me know what you think of these but a command line execution of `test-two-body.py` with `two-body-scf-guess` module imported is `OK` as written.

Reply:
This is better; however, I would recommend only defining variables in once (you have `mol` and `bs` defined twice). Subsequently you should also move the remainder of your global variables either into the `TestTwoBodyGuess` class or into the `if main` block. Once you do that try running the test again; at least for me the test fails.

Reply:
`OK`